### DISCUSSION TITLE
Implementar o operador tipo de

### DISCUSSION THREAD

**Discussion by JohnPetros** [DesignLiquido/delegua#385] [Ideias]
Em JS há o operador `typeof` que indica o tipo de dado de um valor fornecido

```
var nome = "John";
console.log(typeof nome); // Saída: "string"

var idade = 30;
console.log(typeof idade); // Saída: "number"

var ativo = true;
console.log(typeof ativo); // Saída: "boolean"

var lista = [1, 2, 3];
console.log(typeof lista); // Saída: "object"

var pessoa = { nome: "John", idade: 30 };
console.log(typeof pessoa); // Saída: "object"

var funcao = function() {};
console.log(typeof funcao); // Saída: "function"
```

Acho importante implementar esse tipo de recurso porque há momentos na minha aplicação em que é preciso escrever em Delégua que não sei qual tipo de dado está sendo retornado.

O nome em português seria o `tipo de` como sugerido por @leonelsanchesdasilva no Discord.

**Comment by leonelsanchesdasilva** [comment]
@JohnPetros Eu adorei essa ideia. Deve fazer parte do nosso Backlog em breve.

**Comment by samuelrvg** [comment]
@JohnPetros 

A partir da [versão 0.21.0](https://github.com/DesignLiquido/delegua/releases/tag/0.21.0)

![image](https://github.com/DesignLiquido/delegua/assets/14062631/a1d45244-2329-49c7-a7a2-175c892ae654)
